reactive/kubernetes_control_plane.py

```diff
@@ -120,12 +120,34 @@
 ]
 
 cohort_snaps = snap_resources
+CHARM_PREFIX = kubernetes_control_plane.CHARM_PREFIX
+
+
+class MkFlag:
+    """Creates flags which can be used by reactive calls such as `when` or `set_flag`."""
+
+    _prefix = {
+        "charm": CHARM_PREFIX,
+        "nrpe": "nrpe-external-master",  # wokeignore:rule=master
+        "cdk": "cdk.master",  # wokeignore:rule=master
+    }
+
+    def __getattr__(self, item):
+        """Yields a flag making callable for a given _prefix key."""
+        try:
+            pre = self._prefix[item]
+        except KeyError:
+            raise AttributeError(
+                f"'{type(self).__name__}' object has no attribute '{item}'"
+            )
+        return lambda *post: ".".join((pre, *post))
```

Comment:
I think this is a step backward for the understandability of the charm. It's going to slow contributors down if they have to mentally translate between the `flags.charm("upgrade-specified")` that they see in the code, vs. the `kubernetes-master.upgrade-specified` that they see in testing, development, and production. I can't support the loss of clarity here.

Sprinkling `wokeignore` throughout is annoying but I would accept that, more than this, since that would preserve clarity.

Even better would be to rename the flags during charm upgrade so we don't have to dance around them in the code any longer. That's a bit more of an undertaking of course, but I think it could be done.